Backports of upstream commits for gtk-3.20 css compatibility.
Can be dropped when moving to vte 0.43.91 or later.

commit 3e17cbf2d35cd4c64cd8af2944a49407834ab41b
Author: Christian Persch <chpe@gnome.org>
Date:   Thu Jan 14 17:05:33 2016 +0100

    widget: Add CSS name for the terminal widget

commit f34c6fb8e9c795eec6ea993372625b094f2b3c51
Author: Alberts MuktupƒÅvels  <alberts.muktupavels@gmail.com>
Date:   Sat Feb 6 16:16:27 2016 +0200

    vtegtk: use CSS name for widget style
    
    GTK+ now use only CSS name for style. This also changes CSS name
    from vte to vte-terminal to make CSS name similar to type name.

diff --git a/src/vte.cc b/src/vte.cc
index 2b4619c..475297b 100644
--- a/src/vte.cc
+++ b/src/vte.cc
@@ -88,6 +88,8 @@ typedef gunichar wint_t;
 
 #define I_(string) (g_intern_static_string(string))
 
+#define VTE_TERMINAL_CSS_NAME "vte-terminal"
+
 static int _vte_unichar_width(gunichar c, int utf8_ambiguous_width);
 static void vte_terminal_set_visibility (VteTerminal *terminal, GdkVisibilityState state);
 static void vte_terminal_paste(VteTerminal *terminal, GdkAtom board);
@@ -10647,6 +10649,10 @@ vte_terminal_class_init(VteTerminalClass *klass)
 	widget_class->size_allocate = vte_terminal_size_allocate;
         widget_class->screen_changed = vte_terminal_screen_changed;
 
+#if GTK_CHECK_VERSION(3, 19, 5)
+        gtk_widget_class_set_css_name(widget_class, VTE_TERMINAL_CSS_NAME);
+#endif
+
 	/* Initialize default handlers. */
 	klass->eof = NULL;
 	klass->child_exited = NULL;
@@ -11496,7 +11502,7 @@ vte_terminal_class_init(VteTerminalClass *klass)
 
         klass->priv->style_provider = GTK_STYLE_PROVIDER (gtk_css_provider_new ());
         gtk_css_provider_load_from_data (GTK_CSS_PROVIDER (klass->priv->style_provider),
-                                         "VteTerminal {\n"
+                                         "VteTerminal, " VTE_TERMINAL_CSS_NAME " {\n"
                                            "padding: 1px 1px 1px 1px;\n"
                                            "background-color: @theme_base_color;\n"
                                            "color: @theme_fg_color;\n"
